src/main/resources/docker-entrypoint.sh

```diff
@@ -0,0 +1,206 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Requirements:
+# 1) Exec as user
+# * Under Alpine: su-exec (e.g. apk add --no-cache su-exec)
+# * Otherwise: gosu (e.g. apt-get install gosu)
+# 2) Add user and group
+# * Under Alpine: adduser, addgroup
+# * Otherwise: useradd, groupadd
+
+# Exit on any command error
+set -e
+
+# Whether to log during execution
+verbose="true"
+
+# The command and arguments to execute
+command_args=
+
+# The configuration for docker entrypoint
+dep_configuration=
+
+log_err() {
+  l_prefix=$(date  +'%H:%M:%S')
+  printf "[%s] %s\\n" "${l_prefix}" "$@" 1>&2;
+}
+
+log_out() {
+  if [ -n "${verbose}" ]; then
+    l_prefix=$(date  +'%H:%M:%S')
+    printf "[%s] %s\\n" "${l_prefix}" "$@"
+  fi
+}
+
+assert_user_group_matches() {
+  l_name="$1"
+  l_uid="$2"
+  l_gid="$3"
+
+  l_existing_uid=$(id -u "${l_name}")
+  if [ "${l_existing_uid}" != "${l_uid}" ]; then
+    # The user exists but the user id does not match
+    log_err "The user ${l_name} already exists but with uid ${l_existing_uid} and not ${l_uid}"
+    exit 1
+  fi
+
+  l_existing_gid=$(id -g "${l_name}")
+  if [ "${l_existing_gid}" != "${l_gid}" ]; then
+    # The user exists but the group id does not match
+    log_err "The user ${l_name} already exists but with gid ${l_existing_gid} and not ${l_gid}"
+    exit 1
+  fi
+}
+
+linux_create_user_group() {
+  l_name="$1"
+  l_uid="$2"
+  l_gid="$3"
+
+  if id -u "${l_name}" > /dev/null 2>&1; then
+    # Check that the existing user's uid and gid match the given ones
+    assert_user_group_matches "${l_name}" "${l_uid}" "${l_gid}"
+
+    # The user exists with matching user and group id
+    log_out "The user ${l_name} already exists with required ids ${l_existing_uid}:${l_existing_gid}"
+    return
+  fi
+
+  log_out "Creating user ${l_name} (${l_uid}) as member of group ${l_name} (${l_gid})"
+  groupadd -r "${l_name}" "--gid=${l_gid}"
+  useradd -r -g "${l_name}" "--uid=${l_uid}" "${l_name}"
+}
+
+alpine_create_user_group() {
+  l_name="$1"
+  l_uid="$2"
+  l_gid="$3"
+
+  if id -u "${l_name}" > /dev/null 2>&1; then
+    # Check that the existing user's uid and gid match the given ones
+    assert_user_group_matches "${l_name}" "${l_uid}" "${l_gid}"
+
+    # The user exists with matching user and group id
+    log_out "The user ${l_name} already exists with required ids ${l_existing_uid}:${l_existing_gid}"
+    return
+  fi
+
+  log_out "Creating user ${l_name} (${l_uid}) as member of group ${l_name} (${l_gid})"
+  addgroup -g "${l_gid}" -S "${l_name}"
+  adduser -u "${l_uid}" -S -G "${l_name}" "${l_name}"
+}
+
+linux_get_su_cmd() {
+  printf "gosu"
```

Comment:
gosu?

Reply:
Yep.

```
The gosu utility is often used in scripts called from ENTRYPOINT instructions inside Dockerfiles for official images. It's a very simple utility, similar to sudo, that runs a given instruction as a given user. The difference is that gosu avoids sudo's "strange and often annoying TTY and signal-forwarding behavior".
```

https://blog.container-solutions.com/6-dockerfile-tips-official-images#:~:text=The%20gosu%20utility%20is%20often,and%20signal%2Dforwarding%20behavior%22.

Reply:
Any reason to use gosu (that you have to install in the container) instead of su that will already be there?

Reply:
Environment. The docker image sets up an environment that it should run in. Using `su` (even with `-p`, preserve and/or `-w`, whitelist) parts of the environment are reset. This does not happen with `gosu`.

https://man7.org/linux/man-pages/man1/su.1.html